### PR TITLE
Support lightgbm booster model file

### PR DESCRIPTION
**Pull request by caoyi0905**
Support lightgbm booster model file.

Like this:
```
import lightgbm as lgb
from sklearn.datasets import load_iris

X, y = load_iris(return_X_y=True)
estimator = lgb.LGBMClassifier(n_estimators=2, max_depth=1)
estimator.fit(X, y)
estimator.booster_.save_model('lgb_model.txt')
booster = lgb.Booster(model_file='lgb_model.txt')
code = m2c.export_to_java(booster)
```

Due to [No way to differentiate between regression and classification at the Booster level](https://github.com/Microsoft/LightGBM/issues/1700), it will show warning with the booster generated by lightgbm lower than 2.2.3.

### CODE REVIEW

File: m2cgen/assemblers/boosting.py [hunk 1]
```diff
@@ -22,6 +25,14 @@ def __init__(self, model, trees, base_score=0):
             self._is_classification = True
             if model.n_classes_ > 2:
                 self._output_size = model.n_classes_
+        elif type(model) == lgb.basic.Booster:
+            model_dump = model.dump_model()
+            objective = model_dump.get('objective', None)
+            if objective is None:
+                warnings.warn('Please use a model generated by lightgbm which version >= 2.2.3')
```
[the remark this answers is not in the document]
Comment (caoyi0905) [reply m2cgen/assemblers/boosting.py]: It cannot distinguish between Booster from binary LGBMClassifier  and LGBMRegressor in lower version.Actually, it still works on lgbregressor(default _is_classification = False). but i will change this to raise Error.

File: m2cgen/assemblers/__init__.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 from .tree import TreeModelAssembler
 from .ensemble import RandomForestModelAssembler
 from .boosting import XGBoostModelAssembler, LightGBMModelAssembler
+import lightgbm as lgb
```
[the remark this answers is not in the document]
Comment (caoyi0905) [reply m2cgen/assemblers/__init__.py]: How abount change code
```
if type(model) == lgb.basic.Booster:
```
to:
```
if model.__module__ == 'lightgbm.basic' and model.__class__.__name__ == 'Booster'
```

[the remark this answers is not in the document]
Comment (caoyi0905) [reply m2cgen/assemblers/__init__.py]: And I will add test code for this pr.

File: m2cgen/assemblers/boosting.py [hunk 3]
```diff
@@ -22,6 +25,14 @@ def __init__(self, model, trees, base_score=0):
...
+            elif objective.startswith('binary') or objective.startswith('multiclass'):
```
[the remark this answers is not in the document]
Comment (caoyi0905) [reply m2cgen/assemblers/boosting.py]: it generated by [https://github.com/Microsoft/LightGBM/pull/1850/files#diff-40c56d8a449dd067d17653defd8d6331R27](there). it's objective_function ToString funtion.

Possible values:
- "binary sigmoid:xx" : [binary_objective](https://github.com/Microsoft/LightGBM/blob/90127b526b/src/objective/binary_objective.hpp#L164)
- "multiclassova num_class:xx sigmoid:xx" : [multiclass_objective](https://github.com/Microsoft/LightGBM/blob/90127b526b/src/objective/multiclass_objective.hpp#L225)
- "lambdarank" [rank_objective](https://github.com/Microsoft/LightGBM/blob/90127b526b/src/objective/rank_objective.hpp#L200)
- "regression (sqrt)" [regression_objective](https://github.com/Microsoft/LightGBM/blob/90127b526b/src/objective/regression_objective.hpp#L128)
- "xentropy" [xentropy_objective](https://github.com/Microsoft/LightGBM/blob/90127b526b/src/objective/xentropy_objective.hpp#L226)


If objective is "xentropy", I cannot determine it's generated by LGBMRegressor or LGBMClassifier.

**Comment by krinart** [comment]
Please consider following pretty hacky way to do the same, but without any code changes:
```python
import m2cgen as m2c

import lightgbm as lgb
from sklearn.datasets import load_iris

X, y = load_iris(return_X_y=True)
estimator = lgb.LGBMClassifier(n_estimators=2, max_depth=1)
estimator.fit(X, y)
estimator.booster_.save_model('lgb_model.txt')
booster = lgb.Booster(model_file='lgb_model.txt')

# Create fake LGBMClassifier with booster restored from the file.
estimator2 = lgb.LGBMClassifier(n_estimators=2, max_depth=1)
estimator2._Booster = booster
estimator2._n_classes = estimator._n_classes

code = m2c.export_to_java(estimator2)
```

File: tests/assemblers/test_lightgbm.py [hunk 4]
```diff
@@ -12,6 +12,12 @@ def test_binary_classification():
     assembler = assemblers.LightGBMModelAssembler(estimator)
     actual = assembler.assemble()
 
+    estimator.booster_.save_model('lgb_model.txt')
```
Comment (izeigerman) [inline tests/assemblers/test_lightgbm.py]: Where this file is saved? Should we use temporary directory instead? Consider using the `tmp_dir()` context manager from the `tests.utils` package. Like:
```
import os
from tests import utils

with utils.tmp_dir() as tmp_dirpath:
  lab_model_path = os.path.join(tmp_dirpath, 'lgb_model.txt')
  ...
```

**Comment by izeigerman** [comment]
+1 to @krinart 's proposal

**Comment by caoyi0905** [comment]
so i will close this pr~
